src/Response/JSON.php

```diff
@@ -139,6 +139,10 @@ protected function getContent()
      */
     protected static function camelizeObject($source)
     {
+        if (!$source) {
```

Comment:
shouldn't his test be a bit more specific like `if ($source === null)`?

Reply:
I don't disagree. I saw further down in the file that json_last_error() got checked with !$errorCode and not a specific 0 or JSON_ERROR_NONE so I kept my check loose. Many of the values are also strings or booleans, so I think the null check would be best from what I'm seeing if there should be a more specific check.

This actually got me looking though again after I saw you were checking if the $source was an array first before iterating through it...
Maybe I'm looking too far down the path of the code or in addition, in the parse() method, maybe
$this->data   = self::camelizeObject($source);
I should also change to
$this->data = is_array($source) ? self::camelizeObject($source) : $source;
so that only objects get passed.

Reply:
I'm ok with loose checks as well to be consistent. the thing is that I personally don't use the api any more and i can't verify your chance. I'm trusting you that it makes sense.

I'm just thinking what kind of data could be passed to that method

Reply:
That makes sense, and I'm thankful for this conversation as I can better see what you mean. I made a new commit that should better check for the kind of data processed by camelizeObject() and then also only use it in parse() if it's an object to begin with.